### PR TITLE
add support to transcribe Extract function for date, time, datetime and interval

### PR DESCRIPTION
**Pull request by xd1313113**
*Issue #, if available:*

*Description of changes:*
 support to transcribe Extract function for date, time, datetime and interval

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/test/resources/inputs/builtins/extract.sql [hunk 1]
```diff
@@ -0,0 +1,167 @@
+-- Top-level EXTRACT expressions from TIMESTAMP
```
Comment (XuechunHHH) [inline src/test/resources/inputs/builtins/extract.sql]: Have all these tests passed?

File: src/test/resources/inputs/builtins/extract.sql [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+EXTRACT(DAY FROM INTERVAL '10' DAY);
+
+--#[extract-16]
+EXTRACT(HOUR FROM INTERVAL '8' HOUR);
+
+--#[extract-17]
+EXTRACT(MINUTE FROM INTERVAL '30' MINUTE);
+
+--#[extract-18]
+EXTRACT(SECOND FROM INTERVAL '45' SECOND);
```
Comment (yliuuuu) [inline src/test/resources/inputs/builtins/extract.sql]: Could we add a couple cases on Year Month/ Day time? 

```
EXTRACT(MONTH FROM INTERVAL '1-6' YEAR TO MONTH);
```
--------

What do we do on: 
```
EXTRACT(MONTH FROM INTERVAL '18'  MONTH);
```

This is more question on literal transpilation, do we normalize `INTERVAL '18'  MONTH` to `INTERVAL '1-6' YEAR TO MONTH)`? 

--------

Also, seemingly we support negative quantity unit in datetime literal? Do we also want to add some test cases for those for the sake of completeness?

Comment (xd1313113) [reply src/test/resources/inputs/builtins/extract.sql]: Sure, I added more comprehensive tests for interval and negative cases. 

Yes, we do normalize internally. But we do not change type. Test cases are added to interval.sql instead
`INTERVAL '18'  MONTH `-> DatumYM with `1-6` -> transcribe to  `INTERVAL '18'  MONTH `
The bug is fixed with this PR #128
